src/components/Dashboard/Dashboard.jsx

```diff
@@ -8,54 +8,162 @@ import Clock from '../Clock';
 import { useNavigate, useLocation } from 'react-router-dom';
 
 import { PlusIcon } from '@heroicons/react/outline';
+import { CogIcon } from '@heroicons/react/outline';
+
+import { useAuth } from '../../contexts/AuthContextProvider';
+import { query, collection, limit, doc, getDoc, setDoc, onSnapshot, orderBy } from 'firebase/firestore';
+import { db } from '../../utils/FirebaseConfig';
+import AddBookmark from './AddBookmark';
 
 const Dashboard = () => {
+	const [deleteBookmark, setDeleteBookmark] = useState(false);
+	const [addBookmark, setAddBookmark] = useState(false);
 	const navigate = useNavigate();
 	const location = useLocation();
+	const { user } = useAuth();
+	const [bookmarks, setBookmarks] = useState([]);
+
+	const bookmarkCountRef = useRef();
 
 	let params = new URLSearchParams(location.search);
 
 	const handleSettings = () => {
 		navigate('/settings/feedback');
 	};
 
+	useEffect(() => {
+		if (!user) return;
+
+		const queryRef = query(collection(db, 'users', `${user.uid}/bookmarks`), limit(10), orderBy('createdAt', 'desc'));
+
+		const unsubscribe = onSnapshot(queryRef, (snapshot) => {
+			let bookmarksArray = [];
+			snapshot.forEach((doc) => {
+				const bookmarkData = {
+					id: doc.id,
+					url: doc.data().url,
+					title: doc.data().title,
+					favicon: doc.data().favicon,
+				};
+
+				bookmarksArray.push(bookmarkData);
+			});
+			setBookmarks(bookmarksArray);
+		});
+
+		return () => {
+			unsubscribe();
+		};
+	}, [user]);
+
+	useEffect(() => {
+		if (!user) return;
+
+		// recieve bookmark from radial
+		window.api.recieve('bookmarkReply', ([bookmarkData]) => {
+			const queryRefLength = query(collection(db, 'users', `${user.uid}/bookmarks`));
+			onSnapshot(queryRefLength, (snapshot) => {
+				bookmarkCountRef.current = snapshot.size;
+			});
+
+			// add bookmark to firebase db collection
+			const bookmarkRef = doc(db, `users/${user.uid}/bookmarks/${bookmarkData.title}`);
+			// check if bookmark already exists
+			getDoc(bookmarkRef).then((docSnap) => {
+				if (docSnap.exists) {
+					// collection already exists
+					if (docSnap.data()) {
+						// bookmark already exists
+						window.api.send('alert-message-bookmark', {
+							message: 'Bladwijzer bestaat al',
+							type: 'warning',
```

Comment:
Moeten deze zaken via electron gaan? Is het niet mogelijk hier gewoon een react-toast oid te laten zien?

Reply:
Euhm omdat de browserView op de mainwindow wordt gelegd gaat de Toast altijd achter de browserView zitten als het via React wordt gedaan.